### ISSUE TITLE
python manage.py loaddata is killed mysteriously while loading mutations

### ISSUE THREAD

**Issue by kurtwheeler** [cognoma/core-service#57]
While attempting to solve https://github.com/cognoma/core-service/issues/56#issuecomment-305877278 @dhimmel and I ran `python manage.py loaddata` on one of the EC2 instances within the running Docker container. The script got `Killed` mysteriously. Here is the output of said script:

```
root@core-service:/code# python manage.py loaddata                  
Loading mutations table...
Processing 1000 rows so far
Processing 2000 rows so far
Processing 3000 rows so far
Processing 4000 rows so far
Processing 5000 rows so far
Processing 6000 rows so far
Processing 7000 rows so far
Bulk loading mutation data...
Killed
root@core-service:/code# $?
bash: 137: command not found
```

We researched what error code 137 means and it appears to be a SIGKILL with priority 9. We cannot determine what would be sending that signal. @dhimmel thinks it may be caused by running out of memory, however we monitored the memory usage during the execution and it didn't exceed 23% of the memory. We tried multiple times to execute this command, some of which it died before getting as far as it did in the above output.

[This](https://github.com/cognoma/core-service/blob/7f185bc51a73e1cf0571cb2ed601b381f5c49d79/api/management/commands/loaddata.py#L75-L95) is the relevant code block where the command is getting murdered.

We used the API to inspect the number of diseases, samples, and genes and those tables all seem to have been populated successfully.

@awm33 @stephenshank any ideas?

**Comment by dcgoss** [comment]
Haven't looked in depth, but did come across this relevant stack overflow
answer that you may have already seen:
https://stackoverflow.com/questions/19189522/what-does-killed-mean

**Comment by dcgoss** [comment]
After some investigation using [dmesg](https://serverfault.com/a/308899/341225) it appears the culprit is indeed memory. The python process is killed by the [Linux oom-killer](https://linux-mm.org/OOM_Killer), after utilizing 524288/524288 kB of memory. Here are the logs:
![screen shot 2017-06-19 at 1 47 06 pm](https://user-images.githubusercontent.com/7605010/27298279-fc1dc8da-54f5-11e7-871d-dbf99e8b8c9d.png)
![screen shot 2017-06-19 at 1 47 40 pm](https://user-images.githubusercontent.com/7605010/27298281-fc1ff196-54f5-11e7-95bf-a08c330c9bb4.png)
![screen shot 2017-06-19 at 1 47 56 pm](https://user-images.githubusercontent.com/7605010/27298280-fc1e732a-54f5-11e7-91f1-aa79cd34f49a.png)

**Comment by dhimmel** [comment]
Nice forensics! @dcgoss do you want to modify [the source](https://github.com/cognoma/core-service/blob/7f185bc51a73e1cf0571cb2ed601b381f5c49d79/api/management/commands/loaddata.py#L75-L95) to be more memory efficient.

**Comment by dcgoss** [comment]
@dhimmel Yeah I'm checking that out now

**Comment by kurtwheeler** [comment]
I believe those machines should have more memory than that. The containers
may be artificially limited by their task definitions. That is a setting
within AWS' ECS. There is a directory in the infrastructure repo that has
them. Might be worth taking a look at what the settings are.

**Comment by dcgoss** [comment]
Solved. I was doing some intense work on the script and I rewrote the it two times, before discovering that for some reason on the server the `Mutation.objects.bulk_create(mutation_list, batch_size=1000)` line was only `Mutation.objects.bulk_create(mutation_list)`. Django was trying to create all the objects at once, using too much memory. It appears that commit 7f185bc51a73e1cf0571cb2ed601b381f5c49d79 was not deployed to the server.
